### PR TITLE
Regexp

### LINKED ISSUE #3: Regex parser
regex parser
```cpp
Regex parser(const std::string &regex);
bool doesMatch = parser.match(const std::string &str);
```

### PR DESCRIPTION
**Pull request by pmerku**
This resolves #3

### CODE REVIEW

File: src/regex/Regex.cpp [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+	}
+	nodes[j].type = NONE;
+}
+
+Regex::Regex(const std::string &pattern) : _regex() {
+	try {
+		this->compile(pattern);
+	} catch (...) {
+		throw;
+	}
```
Comment (mrjvs) [inline src/regex/Regex.cpp]: useless try catch

File: src/includes/regex/Regex.hpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+			return "The min/max quantifier value is not possible";
+		}
+	};
+
+	class NoTextToMatch : public std::exception {
+	public:
+		const char *what() const throw() {
+			return "The provided text is empty";
+		}
+	};
```
Comment (mrjvs) [inline src/includes/regex/Regex.hpp]: indent them by one, they are under the public label.
also, make them all inherit from a custom exception so you can try catch all regex compilation errors with one catch.
```cpp
class RegexCompilationException: public std::exception {
public:
	virtual const char *what() const throw() {
		return "Failed to compile regex";
	}
};

class NoTextToMatch : public RegexCompilationException {
public:
	const char *what() const throw() {
		return "The provided text is empty";
	}
};

```

**Review by mrjvs — CHANGES_REQUESTED** [review]

File: src/regex/Regex.cpp (hunk 1 above)
Comment (pmerku) [reply src/regex/Regex.cpp]: Removed

File: src/includes/regex/Regex.hpp (hunk 2 above)
Comment (pmerku) [reply src/includes/regex/Regex.hpp]: Fixed

**Review by mrjvs — APPROVED** [review]

**Pull request merged**